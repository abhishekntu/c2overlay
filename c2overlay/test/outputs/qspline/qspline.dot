digraph graphname {
N29 [color="black", label="N19_mul_N19_N29"];
N28 [color="black", label="N18_mul_N18_N28"];
N23 [color="black", label="N11_mul_N11_N23"];
N22 [color="black", label="N10_mul_N10_N22"];
N21 [color="black", label="N13_mul_N13_N21"];
N20 [color="black", label="N12_mul_N12_N20"];
N27 [color="black", label="N15_mul_N15_N27"];
N26 [color="black", label="N14_mul_N14_N26"];
N25 [color="black", label="N17_mul_N17_N25"];
N24 [color="black", label="N16_mul_Imm_4_N16_N24"];
N8 [color="black", label="N29_mul_N29_N8"];
N9 [color="black", label="N28_mul_N28_N9"];
N1 [color="black", label="I0_N1_N1"];
N2 [color="black", label="I1_N2_N2"];
N3 [color="black", label="I2_N3_N3"];
N4 [color="black", label="I3_N4_N4"];
N5 [color="black", label="I4_N5_N5"];
N6 [color="black", label="I5_N6_N6"];
N7 [color="black", label="I6_N7_N7"];
N12 [color="black", label="N21N32_mul_N21_add_N32_N12"];
N13 [color="black", label="N20_mul_N20_N13"];
N10 [color="black", label="N23_mul_N23_N10"];
N11 [color="black", label="N22_mul_Imm_4_N22_N11"];
N16 [color="black", label="N25N33_mul_N25_add_N33_N16"];
N17 [color="black", label="N24_mul_N24_N17"];
N14 [color="black", label="N27_mul_Imm_6_N27_N14"];
N15 [color="black", label="N26N31_mul_N26_add_N31_N15"];
N18 [color="black", label="N8N30_mul_N8_add_N30_N18"];
N19 [color="black", label="N9_mul_N9_N19"];
N30 [color="black", label="O0_N34_N30"];
N29 -> N13;
N28 -> N19;
N23 -> N8;
N22 -> N9;
N22 -> N10;
N21 -> N23;
N20 -> N15;
N27 -> N20;
N26 -> N27;
N25 -> N29;
N24 -> N25;
N8 -> N18;
N9 -> N19;
N1 -> N11;
N2 -> N13;
N2 -> N23;
N2 -> N17;
N2 -> N18;
N2 -> N22;
N2 -> N21;
N2 -> N28;
N2 -> N16;
N2 -> N8;
N3 -> N24;
N4 -> N14;
N5 -> N21;
N6 -> N20;
N6 -> N25;
N6 -> N12;
N6 -> N28;
N6 -> N26;
N6 -> N29;
N6 -> N27;
N6 -> N22;
N6 -> N15;
N7 -> N26;
N12 -> N16;
N13 -> N12;
N10 -> N17;
N11 -> N10;
N16 -> N18;
N17 -> N16;
N14 -> N9;
N15 -> N12;
N18 -> N30;
N19 -> N15;
}
