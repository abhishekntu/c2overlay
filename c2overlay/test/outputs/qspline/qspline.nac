procedure foo(in s32 a,in s32 b,in s32 q,in s32 u,in s32 v,in s32 w,in s32 z,out s32 D_2426)
{
  localvar s32 D_2423;
  localvar s32 D_2424;
  localvar s32 D_2425;
  localvar s32 D_2427;
  localvar s32 D_2428;
  localvar s32 D_2429;
  localvar s32 D_2430;
  localvar s32 D_2431;
  localvar s32 D_2432;
  localvar s32 D_2433;
  localvar s32 D_2434;
  localvar s32 temp_1;
  localvar s32 temp_2;
  localvar s32 temp_3;
  localvar s32 temp_4;
  localvar s32 temp_5;
  localvar s32 temp_6;
  localvar s32 temp_7;
  localvar s32 temp_8;
  localvar s32 temp_9;
  localvar s32 temp_10;
  localvar s32 temp_11;
  localvar s32 temp_12;
  localvar s32 temp_13;
  localvar s32 temp_14;
L0005:
  temp_1 <= mul z,u;
  temp_2 <= mul temp_1,u;
  temp_3 <= mul temp_2,u;
  temp_4 <= mul q,v;
  temp_5 <= mul temp_4,v;
  temp_6 <= mul temp_5,v;
  temp_7 <= mul u,v;
  temp_8 <= mul v,u;
  D_2423 <= mul a,4;
  temp_9 <= mul D_2423,u;
  temp_10 <= mul temp_9,u;
  D_2424 <= mul v,temp_10;
  temp_11 <= mul D_2424,u;
  D_2425 <= mul w,4;
  temp_12 <= mul D_2425,temp_7;
  temp_13 <= mul temp_12,v;
  temp_14 <= mul temp_13,v;
  D_2427 <= mul u,temp_3;
  D_2428 <= mul b,6;
  D_2429 <= mul D_2428,temp_7;
  D_2430 <= mul D_2429,temp_8;
  D_2431 <= add D_2427,D_2430;
  D_2432 <= add D_2431,temp_11;
  D_2433 <= add D_2432,temp_14;
  D_2434 <= mul v,temp_6;
  D_2426 <= add D_2433,D_2434;
}
