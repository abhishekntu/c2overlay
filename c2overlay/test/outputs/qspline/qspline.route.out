Array size: 5 x 5 logic blocks.

Routing:

Net 0 (N1)

SOURCE (0,2)  Pad: 1  
  OPIN (0,2)  Pad: 1  
 CHANY (0,2)  Track: 1  
  IPIN (1,2)  Pin: 3  
  SINK (1,2)  Class: 0  


Net 1 (N2)

SOURCE (3,6)  Pad: 1  
  OPIN (3,6)  Pad: 1  
 CHANX (3,5)  Track: 0  
 CHANX (4,5)  Track: 0  
  IPIN (4,5)  Pin: 0  
  SINK (4,5)  Class: 0  
 CHANX (4,5)  Track: 0  
 CHANY (4,5)  Track: 1  
 CHANY (4,4)  Track: 1  
  IPIN (4,4)  Pin: 1  
  SINK (4,4)  Class: 0  
 CHANY (4,4)  Track: 1  
 CHANX (4,3)  Track: 1  
 CHANX (3,3)  Track: 1  
  IPIN (3,4)  Pin: 2  
  SINK (3,4)  Class: 0  
 CHANX (3,3)  Track: 1  
 CHANX (2,3)  Track: 1  
  IPIN (2,4)  Pin: 2  
  SINK (2,4)  Class: 0  
  OPIN (3,6)  Pad: 1  
 CHANX (3,5)  Track: 1  
 CHANX (2,5)  Track: 1  
  IPIN (2,5)  Pin: 0  
  SINK (2,5)  Class: 0  
 CHANX (2,3)  Track: 1  
 CHANX (1,3)  Track: 1  
  IPIN (1,4)  Pin: 2  
  SINK (1,4)  Class: 0  
 CHANX (2,3)  Track: 1  
  IPIN (2,3)  Pin: 0  
  SINK (2,3)  Class: 0  
 CHANX (2,3)  Track: 1  
 CHANY (1,3)  Track: 1  
 CHANY (1,2)  Track: 1  
  IPIN (2,2)  Pin: 3  
  SINK (2,2)  Class: 0  
 CHANX (4,3)  Track: 1  
 CHANY (3,3)  Track: 1  
 CHANY (3,2)  Track: 1  
  IPIN (4,2)  Pin: 3  
  SINK (4,2)  Class: 0  


Net 2 (N3)

SOURCE (1,0)  Pad: 1  
  OPIN (1,0)  Pad: 1  
 CHANX (1,0)  Track: 1  
  IPIN (1,1)  Pin: 2  
  SINK (1,1)  Class: 0  


Net 3 (N4)

SOURCE (3,0)  Pad: 1  
  OPIN (3,0)  Pad: 1  
 CHANX (3,0)  Track: 0  
  IPIN (3,1)  Pin: 2  
  SINK (3,1)  Class: 0  


Net 4 (N5)

SOURCE (0,4)  Pad: 1  
  OPIN (0,4)  Pad: 1  
 CHANY (0,4)  Track: 1  
  IPIN (1,4)  Pin: 3  
  SINK (1,4)  Class: 0  


Net 5 (N6)

SOURCE (4,0)  Pad: 1  
  OPIN (4,0)  Pad: 1  
 CHANX (4,0)  Track: 1  
  IPIN (4,1)  Pin: 2  
  SINK (4,1)  Class: 0  
  OPIN (4,0)  Pad: 1  
 CHANX (4,0)  Track: 0  
 CHANX (5,0)  Track: 0  
  IPIN (5,1)  Pin: 2  
  SINK (5,1)  Class: 0  
 CHANX (5,0)  Track: 0  
 CHANY (5,1)  Track: 0  
 CHANY (5,2)  Track: 0  
  IPIN (5,2)  Pin: 1  
  SINK (5,2)  Class: 0  
 CHANY (5,2)  Track: 0  
 CHANY (5,3)  Track: 0  
  IPIN (5,3)  Pin: 1  
  SINK (5,3)  Class: 0  
 CHANY (5,2)  Track: 0  
 CHANX (5,2)  Track: 1  
 CHANX (4,2)  Track: 1  
  IPIN (4,3)  Pin: 2  
  SINK (4,3)  Class: 0  
 CHANY (5,3)  Track: 0  
 CHANY (5,4)  Track: 0  
  IPIN (5,4)  Pin: 1  
  SINK (5,4)  Class: 0  
 CHANX (4,0)  Track: 1  
 CHANX (3,0)  Track: 1  
 CHANX (2,0)  Track: 1  
  IPIN (2,1)  Pin: 2  
  SINK (2,1)  Class: 0  
 CHANX (3,0)  Track: 1  
 CHANY (2,1)  Track: 0  
 CHANY (2,2)  Track: 0  
  IPIN (2,2)  Pin: 1  
  SINK (2,2)  Class: 0  
 CHANY (2,2)  Track: 0  
 CHANY (2,3)  Track: 0  
  IPIN (2,3)  Pin: 1  
  SINK (2,3)  Class: 0  


Net 6 (N7)

SOURCE (6,1)  Pad: 1  
  OPIN (6,1)  Pad: 1  
 CHANY (5,1)  Track: 1  
  IPIN (5,1)  Pin: 1  
  SINK (5,1)  Class: 0  


Net 7 (N29)

SOURCE (4,1)  Class: 1  
  OPIN (4,1)  Pin: 4  
 CHANX (4,1)  Track: 1  
  IPIN (4,2)  Pin: 2  
  SINK (4,2)  Class: 0  


Net 8 (N28)

SOURCE (2,3)  Class: 1  
  OPIN (2,3)  Pin: 5  
 CHANY (2,3)  Track: 1  
  IPIN (3,3)  Pin: 3  
  SINK (3,3)  Class: 0  


Net 9 (N23)

SOURCE (2,4)  Class: 1  
  OPIN (2,4)  Pin: 4  
 CHANX (2,4)  Track: 1  
  IPIN (2,5)  Pin: 2  
  SINK (2,5)  Class: 0  


Net 10 (N22)

SOURCE (2,2)  Class: 1  
  OPIN (2,2)  Pin: 5  
 CHANY (2,2)  Track: 1  
  IPIN (3,2)  Pin: 3  
  SINK (3,2)  Class: 0  
SOURCE (2,2)  Class: 1  
  OPIN (2,2)  Pin: 7  
 CHANY (1,2)  Track: 0  
 CHANY (1,3)  Track: 0  
  IPIN (1,3)  Pin: 1  
  SINK (1,3)  Class: 0  


Net 11 (N21)

SOURCE (1,4)  Class: 1  
  OPIN (1,4)  Pin: 5  
 CHANY (1,4)  Track: 1  
  IPIN (2,4)  Pin: 3  
  SINK (2,4)  Class: 0  


Net 12 (N20)

SOURCE (5,4)  Class: 1  
  OPIN (5,4)  Pin: 6  
 CHANX (5,3)  Track: 1  
  IPIN (5,3)  Pin: 0  
  SINK (5,3)  Class: 0  


Net 13 (N27)

SOURCE (5,2)  Class: 1  
  OPIN (5,2)  Pin: 7  
 CHANY (4,2)  Track: 0  
 CHANY (4,3)  Track: 0  
 CHANY (4,4)  Track: 0  
  IPIN (5,4)  Pin: 3  
  SINK (5,4)  Class: 0  


Net 14 (N26)

SOURCE (5,1)  Class: 1  
  OPIN (5,1)  Pin: 4  
 CHANX (5,1)  Track: 1  
  IPIN (5,2)  Pin: 2  
  SINK (5,2)  Class: 0  


Net 15 (N25)

SOURCE (2,1)  Class: 1  
  OPIN (2,1)  Pin: 4  
 CHANX (2,1)  Track: 0  
 CHANX (3,1)  Track: 0  
 CHANX (4,1)  Track: 0  
  IPIN (4,1)  Pin: 0  
  SINK (4,1)  Class: 0  


Net 16 (N24)

SOURCE (1,1)  Class: 1  
  OPIN (1,1)  Pin: 5  
 CHANY (1,1)  Track: 1  
  IPIN (2,1)  Pin: 3  
  SINK (2,1)  Class: 0  


Net 17 (N8)

SOURCE (2,5)  Class: 1  
  OPIN (2,5)  Pin: 5  
 CHANY (2,5)  Track: 1  
 CHANX (3,4)  Track: 0  
 CHANX (4,4)  Track: 0  
  IPIN (4,5)  Pin: 2  
  SINK (4,5)  Class: 0  


Net 18 (N9)

SOURCE (3,2)  Class: 1  
  OPIN (3,2)  Pin: 4  
 CHANX (3,2)  Track: 1  
  IPIN (3,3)  Pin: 2  
  SINK (3,3)  Class: 0  


Net 19 (N12)

SOURCE (4,3)  Class: 1  
  OPIN (4,3)  Pin: 4  
 CHANX (4,3)  Track: 0  
  IPIN (4,4)  Pin: 2  
  SINK (4,4)  Class: 0  


Net 20 (N13)

SOURCE (4,2)  Class: 1  
  OPIN (4,2)  Pin: 7  
 CHANY (3,2)  Track: 0  
 CHANY (3,3)  Track: 0  
  IPIN (4,3)  Pin: 3  
  SINK (4,3)  Class: 0  


Net 21 (N10)

SOURCE (1,3)  Class: 1  
  OPIN (1,3)  Pin: 4  
 CHANX (1,3)  Track: 0  
 CHANX (2,3)  Track: 0  
 CHANY (2,4)  Track: 0  
  IPIN (3,4)  Pin: 3  
  SINK (3,4)  Class: 0  


Net 22 (N11)

SOURCE (1,2)  Class: 1  
  OPIN (1,2)  Pin: 4  
 CHANX (1,2)  Track: 1  
  IPIN (1,3)  Pin: 2  
  SINK (1,3)  Class: 0  


Net 23 (N16)

SOURCE (4,4)  Class: 1  
  OPIN (4,4)  Pin: 4  
 CHANX (4,4)  Track: 1  
 CHANY (3,5)  Track: 0  
  IPIN (4,5)  Pin: 3  
  SINK (4,5)  Class: 0  


Net 24 (N17)

SOURCE (3,4)  Class: 1  
  OPIN (3,4)  Pin: 5  
 CHANY (3,4)  Track: 1  
  IPIN (4,4)  Pin: 3  
  SINK (4,4)  Class: 0  


Net 25 (N14)

SOURCE (3,1)  Class: 1  
  OPIN (3,1)  Pin: 4  
 CHANX (3,1)  Track: 1  
  IPIN (3,2)  Pin: 2  
  SINK (3,2)  Class: 0  


Net 26 (N15)

SOURCE (5,3)  Class: 1  
  OPIN (5,3)  Pin: 7  
 CHANY (4,3)  Track: 1  
  IPIN (4,3)  Pin: 1  
  SINK (4,3)  Class: 0  


Net 27 (N30)

SOURCE (4,5)  Class: 1  
  OPIN (4,5)  Pin: 4  
 CHANX (4,5)  Track: 1  
  IPIN (4,6)  Pad: 0  
  SINK (4,6)  Pad: 0  


Net 28 (N19)

SOURCE (3,3)  Class: 1  
  OPIN (3,3)  Pin: 6  
 CHANX (3,2)  Track: 0  
 CHANX (4,2)  Track: 0  
 CHANX (5,2)  Track: 0  
  IPIN (5,3)  Pin: 2  
  SINK (5,3)  Class: 0  
